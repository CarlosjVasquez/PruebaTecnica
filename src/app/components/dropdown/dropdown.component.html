<div class="dropdown">
  @if (icon) {
  <custom-icon class="dropdown-toggle" [src]="icon" [size]="size" />
  } @if (label) {
  <span class="dropdown-toggle">{{ label }}</span>
  }

  <ul class="dropdown-menu">
    @for (option of options; track $index) {
    <li (click)="onClick(option.id)">
      <span> {{ option.label }} </span>
    </li>
    }
  </ul>
</div>
