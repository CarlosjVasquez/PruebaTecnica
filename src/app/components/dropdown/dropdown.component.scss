/* Estilos generales del dropdown */
.dropdown {
  position: relative;
  display: inline-block;
}

/* Estilos del botón que activa el dropdown */
.dropdown-toggle {
  border: none;
  cursor: pointer;
}

/* Estilos del menú desplegable */
.dropdown-menu {
  position: absolute;
  width: 150px;
  top: 0;
  left: -150px;
  background-color: #f1f1f1;
  color: #333;
  padding: 10px;
  border: 1px solid #ccc;
  border-radius: 5px;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: none;
  z-index: 1;
}

.dropdown-menu li {
  padding: 10px;
  cursor: pointer;
}

.dropdown-menu li a {
  color: #333;
  text-decoration: none;
  display: block;
  font-size: 0.8rem;
}

/* Mostrar el menú cuando se hace hover sobre el botón */
.dropdown:hover .dropdown-menu {
  display: block;
}
