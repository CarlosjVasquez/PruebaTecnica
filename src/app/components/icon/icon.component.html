<picture>
  <img [src]="src" alt="icon" [width]="size" [height]="size" />
</picture>
