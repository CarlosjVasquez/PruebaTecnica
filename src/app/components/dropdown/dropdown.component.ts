import { Component, EventEmitter, Input, Output } from '@angular/core';
import { IconComponent } from '../icon/icon.component';

export interface Option {
  id: number;
  label: string;
}

@Component({
  selector: 'dropdown',
  standalone: true,
  imports: [IconComponent],
  templateUrl: './dropdown.component.html',
  styleUrl: './dropdown.component.scss',
})
export class DropdownComponent {
  @Input() icon: string = '';
  @Input() size: number = 15;
  @Input() label?: string = '';
  @Input() options: Option[] = [];
  @Output() onClickEventEmitter: EventEmitter<number> = new EventEmitter();

  onClick(id: number) {
    this.onClickEventEmitter.emit(id);
  }
}
