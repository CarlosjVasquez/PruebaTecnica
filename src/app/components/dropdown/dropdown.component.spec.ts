import { ComponentFixture, TestBed } from '@angular/core/testing';

import { DropdownComponent } from './dropdown.component';
import { IconComponent } from '../icon/icon.component';
import { By } from '@angular/platform-browser';

describe('DropdownComponent', () => {
  let component: DropdownComponent;
  let fixture: ComponentFixture<DropdownComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      providers: [DropdownComponent, IconComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(DropdownComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('match to snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  test('should icon', () => {
    component.icon = 'assets/icons/navegacion.png';
    fixture.detectChanges();
    const icon = fixture.debugElement.query(By.directive(IconComponent));
    expect(icon).toBeTruthy();
  });

  test('should emit event click', () => {
    jest.spyOn(component.onClickEventEmitter, 'emit');

    component.options = [
      {
        label: 'Option 1',
        id: 1,
      },
    ];
    fixture.detectChanges();
    const liElement = fixture.debugElement.query(By.css('li')).nativeElement;
    liElement.click();

    expect(component.onClickEventEmitter.emit).toHaveBeenCalled();
  });
});
