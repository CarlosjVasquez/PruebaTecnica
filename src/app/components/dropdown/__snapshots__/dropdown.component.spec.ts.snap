// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownComponent match to snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="dropdown"
  >
    
    
    <ul
      class="dropdown-menu"
    >
      
    </ul>
  </div>
</div>
`;
